spec/controllers/users_controller_spec.rb

```diff
@@ -31,73 +31,117 @@
+            invitee = User.find_by(email: "joe@doe.com")
+            expect(invitee.follows?(invitation.inviter)).to be true
+          end
+
+          it "deletes the invitation token" do
+            invitation = Fabricate :invitation, invitee_email: 'joe@doe.com', inviter: Fabricate(:user)
+            post :create, invitation_token: invitation.token, user: { fullname: invitation.invitee_name, email: invitation.invitee_email, password: "newPwd" }
+            invitee = User.find_by(email: "joe@doe.com")
+            expect(invitation.reload.token).to be_nil
+          end
+        end
       end
 
-      context "with invitation" do
-        it "sets the invited user to follow the inviter" do
-          invitation = Fabricate :invitation, invitee_email: 'joe@doe.com', inviter: Fabricate(:user)
-          post :create, invitation_token: invitation.token, user: { fullname: invitation.invitee_name, email: invitation.invitee_email, password: "newPwd" }
-          invitee = User.find_by(email: "joe@doe.com")
-          expect(invitation.inviter.follows?(invitee)).to be true
+      context "but with an invalid credit card" do
+        let(:charge) do
+          double(:charge,
+                 successful?: false,
+                error_message: "Your card was declined.")
         end
 
-        it "sets the inviter to follow the invited user" do
-          invitation = Fabricate :invitation, invitee_email: 'joe@doe.com', inviter: Fabricate(:user)
-          post :create, invitation_token: invitation.token, user: { fullname: invitation.invitee_name, email: invitation.invitee_email, password: "newPwd" }
-          invitee = User.find_by(email: "joe@doe.com")
-          expect(invitee.follows?(invitation.inviter)).to be true
+        before do
+          post :create, user: Fabricate.attributes_for(:user)
         end
 
-        it "deletes the invitation token" do
-          invitation = Fabricate :invitation, invitee_email: 'joe@doe.com', inviter: Fabricate(:user)
-          post :create, invitation_token: invitation.token, user: { fullname: invitation.invitee_name, email: invitation.invitee_email, password: "newPwd" }
-          invitee = User.find_by(email: "joe@doe.com")
-          expect(invitation.reload.token).to be_nil
+        it "does not create a new user" do
+          expect(User.count).to eq 0
+        end
+
+        it "renders the sign up form again" do
+          expect(response).to render_template(:new)
+        end
+
+        it "sets the flash error message" do
+          expect(flash[:danger]).to eq "Your card was declined."
         end
       end
     end
 
-    context "with invalid credentials" do
+    context "with invalid personal information" do
       before do
         post :create, user: {fullname: 'P', password: 'pwd', email: 'a@b.com'}
       end
 
       it "sets @user to a new User" do
+        post :create, user: {fullname: 'P', password: 'pwd', email: 'a@b.com'}
```

Comment:
I found the highly shortened values here made it harder to read. My personal preference here would be: `fullname: 'IDontNeedNoPassword', password '', email: 'idontneednopassword@doe.com'`. It isn't really obvious here why the information is invalid. But also (contrary to what might seem) our brains work better with more complete information. It is the same reasoning why we would use `alice` and `bob` instead of `a` and `b`. Having a tag (even making it a silly tag) helps the process of absorbing the text.